examples/shader/mesh_instance_index.rs

```diff
@@ -0,0 +1,90 @@
+//! This examples demonstrates using the `MeshInstanceIndex` component to sample from a
```

Comment:
I wonder if this should just be merged with the automatic_instancing example?

Just add a few comments to explain that the new component is optional.

Reply:
This one is a lot prettier though, so I'm happy if it keeps the look of it. Just also add the comments explaining the automatic instancing bits.

Reply:
I think that's a good idea actually, as it will be more discoverable and is often what users are curious about when they reach for instancing.